Source/SwiftLintFramework/Rules/SwitchCaseAlignmentRule.swift

```diff
@@ -91,18 +37,163 @@ public struct SwitchCaseAlignmentRule: ASTRule, ConfigurationProviderRule {
 
         let caseLocations = caseStatements.compactMap { caseDict -> Location? in
             guard let byteOffset = caseDict.offset,
-                let (line, char) = contents.lineAndCharacter(forByteOffset: byteOffset) else {
-                    return nil
+                  let (line, char) = contents.lineAndCharacter(forByteOffset: byteOffset) else {
+                return nil
             }
+
             return Location(file: file.path, line: line, character: char)
         }
 
+        guard let firstCase = caseLocations.first,
+              let firstCaseCharacter = firstCase.character else {
+            return []
+        }
+
+        // If indent_cases is on, the first case should be indented from its containing switch.
+        if configuration.indentedCases, firstCaseCharacter <= switchCharacter {
+            return caseLocations.map(locationToViolation)
+        }
+
+        let indentation = configuration.indentedCases ? firstCaseCharacter - switchCharacter : 0
+
         return caseLocations
-            .filter { $0.character != switchCharacter }
-            .map {
-                StyleViolation(ruleDescription: type(of: self).description,
-                               severity: configuration.severity,
-                               location: $0)
-            }
+            .filter { $0.character != switchCharacter + indentation }
+            .map(locationToViolation)
+    }
+
+    private func locationToViolation(_ location: Location) -> StyleViolation {
+        // swiftlint:disable line_length
```

Comment:
How about this to avoid having to disable the line length rule?

```swift
private func locationToViolation(_ location: Location) -> StyleViolation {
    let reason = """
        Case statements should \
        \(configuration.indentedCases ? "be indented within" : "vertically align with") \
        their enclosing switch statement
        """

    return StyleViolation(ruleDescription: SwitchCaseAlignmentRule.description,
                          severity: configuration.severityConfiguration.severity,
                          location: location,
                          reason: reason)
}
```